docs/changelog/133752.yaml

```diff
@@ -0,0 +1,5 @@
+pr: 133752
+summary: Fix enrich fails to update when source changes
```

Comment:
Nit: should we rephrase this to make it easier to understand and find for users? Perhaps something like: `Avoid stale enrich cache results after policy is re-ran`?

Reply:
👍, https://github.com/elastic/elasticsearch/pull/133752/commits/253425b7dc4a5214add7a73232fc46b1ba7e36fd